routes/articles.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express');
+const Artciles = require('../db/articles.js');
+const articles = new Artciles();
+const router = express.Router();
+
+router.post('/', (req, res) => {
+  const data = req.body;
+  if((!data.title || !data.body || !data.author || articles.validate(data.title)) && articles.all().length > 0){
+    res.redirect('/articles/new');
+  }else{
+    articles.create(data);
+    res.redirect('/articles');
+  }
+});
+
+router.put('/:title', (req, res) => {
+  const data = req.body
+  const title = req.params.title;
+  console.log(data);
+  console.log(title);
+  let targetArticle = articles.findArticle(title);
+  if(targetArticle){
+    if(data.title || data.body || data.author){
```

Comment:
I would advise moving more of this logic into the db section. For editing, you should be able to just pass the new value and not deal with getting the current instance of the article.